src/controller/python/README.md

```diff
@@ -1,244 +1,255 @@
+chip-device-ctrl > connect -ble 3840 20202021 1234
 ```
 
-3.  Set wifi credential
+You may skip providing the last parameter which is a Node ID. In such case, the controller will randomly assign it. However, note the Node ID down, because it is required later in the configuration process.
 
-> Note: This command will be deprerated after the network provisioning cluster
-> is ready.
+At the end of the secure connection establishment, the Python controller will indicate this by printing the following log:
 
 ```
-chip-device-ctrl > set-pairing-wifi-credential TestAP TestPassword
+Secure Session to Device Established
 ```
 
-4.  Connect to device using setup pin code
+This means that the PASE (Password-Authenticated Session Establishment) session using SPAKE2+ protocol is completed.
 
-```
-chip-device-ctrl > connect -ble 1383 20202021
-```
+### 5. Commission Matter accessory to the underlying network
 
-## Thread provisioning
+One of the commissioning steps is Network Commissioning. The main goal of this step is to configure a network interface, such as Thread or Wifi, on the device and to provide it with network credentials.
 
-1. Configure Thread border router. For example, follow
-   [Setup OpenThread Border Router on Raspberry Pi / ubuntu](../../../docs/guides/openthread_border_router_pi.md)
-   instruction to configure OpenThread Border Router on a Linux workstation.
+#### Commissioning of Thread device
 
-2. Run CHIP Device Controller
+1. Fetch and store the current Active Operational Dataset and Extended PAN ID from the Thread Border Router.
 
-```
-sudo chip-device-ctrl
-```
+   Depending if Thread Border Router is running on Docker or natively on Raspberry Pi, execute the following commands:
 
-3. Set Thread credentials
+   For Docker:
+   ```
+   sudo docker exec -it otbr sh -c "sudo ot-ctl dataset active -x"
+   0e080000000000010000000300001335060004001fffe002084fe76e9a8b5edaf50708fde46f999f0698e20510d47f5027a414ffeebaefa92285cc84fa030f4f70656e5468726561642d653439630102e49c0410b92f8c7fbb4f9f3e08492ee3915fbd2f0c0402a0fff8
+   Done
 
-```
-set-pairing-thread-credential <channel> <pan id[HEX]> <master_key>
-```
+   sudo docker exec -it otbr sh -c "sudo ot-ctl dataset extpanid”
+   4fe76e9a8b5edaf5
+   Done
+   ```
 
-4. BLE Connect to the device
+   For native installation:
 
-```
-connect -ble <discriminator> <setup pin code> [<nodeid>]
-```
+   ```
+   sudo ot-ctl dataset active -x
+   0e080000000000010000000300001335060004001fffe002084fe76e9a8b5edaf50708fde46f999f0698e20510d47f5027a414ffeebaefa92285cc84fa030f4f70656e5468726561642d653439630102e49c0410b92f8c7fbb4f9f3e08492ee3915fbd2f0c0402a0fff8
+   Done
+
+   sudo ot-ctl dataset extpanid
+   4fe76e9a8b5edaf5
+   Done
+   ```
+
+   Matter specifiction does not define how the Thread or Wi-Fi credentials are obtained by Controller. For example, for Thread, instead of fetching datasets directly from the Thread Border Router, you may also use different out-of-band method.
```

Comment:
```suggestion
   Matter specifiction does not define how the Thread or Wi-Fi credentials are obtained by Controller. For example, for Thread, instead of fetching datasets directly from the Thread Border Router, you might also use a different out-of-band method.
```